### PR TITLE
Adding vanilla prometheus /metrics endpoint

### PR DESCRIPTION
**Pull request by pearj**

### CODE REVIEW

File: src/main/java/de/zalando/ep/zalenium/aspect/HubAspect.aj [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    before(ServletContextHandler handler, Hub hub) : callAddDefaultServlets(handler, hub) {
+        log.info("Registering custom Zalenium servlets");
+        
+        FilterHolder prometheusMetricsFilter = handler.addFilter(MetricsFilter.class, "/*", EnumSet.allOf(DispatcherType.class));
+        prometheusMetricsFilter.setInitParameter("metric-name", "webapp_metrics_filter");
+        prometheusMetricsFilter.setInitParameter("help", "This is the help for your metrics filter");
+        prometheusMetricsFilter.setInitParameter("buckets", "0.005,0.01,0.025,0.05,0.075,0.1,0.25,0.5,0.75,1,2.5,5,7.5,10");
+        
+        // Enable prometheus metrics at /metrics
+        handler.addServlet(new ServletHolder(new MetricsServlet()), "/metrics");
```
Comment (dspasojevic) [inline src/main/java/de/zalando/ep/zalenium/aspect/HubAspect.aj:37]: What will this do is selenium has a `/metrics`? Can we / should we check that?

Comment (pearj) [reply src/main/java/de/zalando/ep/zalenium/aspect/HubAspect.aj:37]: Have a look in Hub.class in selenium. That’s where it registers all it’s servlets. 
I don’t recall seeing a metrics. Plus mine is registered first so I win ;)

Not at my pc, so can’t check right now.

File: src/main/java/de/zalando/ep/zalenium/aspect/HubAspect.aj [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import io.prometheus.client.exporter.MetricsServlet;
+import io.prometheus.client.filter.MetricsFilter;
+import io.prometheus.client.hotspot.DefaultExports;
+import io.prometheus.client.jetty.JettyStatisticsCollector;
+
+privileged public aspect HubAspect {
+    private static final Logger log = LoggerFactory.getLogger(HubAspect.class.getName());
+
+    pointcut callAddDefaultServlets(ServletContextHandler handler, Hub hub) :
+        call (private void Hub.addDefaultServlets(ServletContextHandler)) && args(handler) && target(hub);
```
Comment (pearj) [inline src/main/java/de/zalando/ep/zalenium/aspect/HubAspect.aj:26]: This is where selenium registers all it’s servlets.

File: src/main/java/de/zalando/ep/zalenium/aspect/HubAspect.aj (hunk 1 above)
Comment (pearj) [reply src/main/java/de/zalando/ep/zalenium/aspect/HubAspect.aj:37]: No `/metrics` endpoint:

https://github.com/SeleniumHQ/selenium/blob/157703a475f8061a57deb489b746757e7efc0040/java/server/src/org/openqa/grid/web/Hub.java#L123-L162

**Comment by pearj** [comment]
As discussed, I've pushed up a pull request to the main zalenium repo, and you could rebase off that one.

https://github.com/zalando/zalenium/pull/533

**Pull request closed without merging**